src/streaming/ManifestLoader.js

```diff
@@ -139,111 +139,68 @@ function ManifestLoader() {
-                        if (responseURL && responseURL !== url) {
-                            baseUri = urlUtils.parseBaseUrl(responseURL);
-                            actualUrl = responseURL;
-                        } else {
-                            // usually this case will be caught and resolved by
-                            // responseURL above but it is not available for IE11 and Edge/12 and Edge/13
-                            // baseUri must be absolute for BaseURL resolution later
-                            if (urlUtils.isRelative(url)) {
-                                url = urlUtils.resolve(url, window.location.href);
-                            }
-        
-                            baseUri = urlUtils.parseBaseUrl(url);
-                        }
-        
-                        // A response of no content implies in-memory is properly up to date
-                        if (textStatus == 'No Content') {
-                            eventBus.trigger(
-                                Events.INTERNAL_MANIFEST_LOADED, {
-                                    manifest: null
-                                }
-                            );
-                            return;
-                        }
-        
-                        // Create parser according to manifest type
-                        if (parser === null) {
-                            parser = createParser(data);
-                        }
-        
-                        if (parser === null && !linkPeriod) {
-                            eventBus.trigger(Events.INTERNAL_MANIFEST_LOADED, {
-                                manifest: null,
-                                error: new DashJSError(
-                                    Errors.MANIFEST_LOADER_PARSING_FAILURE_ERROR_CODE,
-                                    Errors.MANIFEST_LOADER_PARSING_FAILURE_ERROR_MESSAGE + `${url}`
-                                )
-                            });
-                            return;
-                        }
-        
-                        // init xlinkcontroller with created parser
-                        xlinkController.setParser(parser);
-        
-                        try {
-                            manifest = parser.parse(data);
-                        } catch (e) { 
-                            if (!linkPeriod) {
-                                eventBus.trigger(Events.INTERNAL_MANIFEST_LOADED, {
-                                    manifest: null,
-                                    error: new DashJSError(
-                                        Errors.MANIFEST_LOADER_PARSING_FAILURE_ERROR_CODE,
-                                        Errors.MANIFEST_LOADER_PARSING_FAILURE_ERROR_MESSAGE + `${url}`
-                                    )
-                                });
-                            }
-                            return;
+        function loadUrl(resolve = null, reject = null) {
```

Comment:
I think that we should move the loadUrl function declaration outside of the load function.

Reply:
Also `load` and `loadUrl` aren’t very descriptive.